server/src/test/java/org/elasticsearch/action/admin/indices/settings/get/GetSettingsResponseTests.java

```diff
@@ -0,0 +1,191 @@
+
+public class GetSettingsResponseTests extends AbstractStreamableXContentTestCase<GetSettingsResponse> {
+
+    private final Set<String> indexNames;
+    public GetSettingsResponseTests() {
+        Set<String> names = new HashSet<String>();
+        int numIndices = randomIntBetween(1, 5);
+        for (int x=0;x<numIndices;x++) {
+            String indexName = randomAlphaOfLength(5);
+            names.add(indexName);
+        }
+        indexNames = Collections.unmodifiableSet(names);
+    }
+    /*
+    index.number_of_shards=2,index.number_of_replicas=1.  The below base64'd bytes were generated by
+    code from the 6.2.2 tag.
+     */
+    private static final String TEST_6_2_2_RESPONSE_BYTES =
+        "AQppbmRleF9uYW1lAhhpbmRleC5udW1iZXJfb2ZfcmVwbGljYXMAATEWaW5kZXgubnVtYmVyX29mX3NoYXJkcwABMg==";
+
+    /* This response object was generated using similar code to the code used to create the above bytes */
+    private static final GetSettingsResponse TEST_6_2_2_RESPONSE_INSTANCE = getExpectedTest622Response();
+
+    @Override
+    protected GetSettingsResponse createBlankInstance() {
+        return new GetSettingsResponse();
+    }
+
+    @Override
+    protected GetSettingsResponse createTestInstance() {
+        HashMap<String, Settings> indexToSettings = new HashMap<>();
+        HashMap<String, Settings> indexToDefaultSettings = new HashMap<>();
+
+        IndexScopedSettings indexScopedSettings = IndexScopedSettings.DEFAULT_SCOPED_SETTINGS;
+
+        for (String indexName : indexNames) {
+            Settings.Builder builder = Settings.builder();
+            builder.put(RandomCreateIndexGenerator.randomIndexSettings());
+            /*
+            We must ensure that *something* is in the settings response as we optimize away empty settings
+            blocks in x content responses
+             */
+            builder.put("index.refresh_interval", "1s");
+            indexToSettings.put(indexName, builder.build());
+        }
+        ImmutableOpenMap<String, Settings> immutableIndexToSettings =
+            ImmutableOpenMap.<String, Settings>builder().putAll(indexToSettings).build();
+
+
+        if (randomBoolean()) {
+            for (String indexName : indexToSettings.keySet()) {
+                Settings defaultSettings = indexScopedSettings.diff(indexToSettings.get(indexName), Settings.EMPTY);
+                indexToDefaultSettings.put(indexName, defaultSettings);
+            }
+        }
+
+        ImmutableOpenMap<String, Settings> immutableIndexToDefaultSettings =
+            ImmutableOpenMap.<String, Settings>builder().putAll(indexToDefaultSettings).build();
+
+        return new GetSettingsResponse(immutableIndexToSettings, immutableIndexToDefaultSettings);
+    }
+
+    @Override
+    protected GetSettingsResponse doParseInstance(XContentParser parser) throws IOException {
+        return GetSettingsResponse.fromXContent(parser);
+    }
+
+    protected boolean supportsUnknownFields() {
+        return false;
```

Comment:
this should be set to true, and we should exclude the index level objects from it. I thought it was that way before?